### PR TITLE
Support multiple files in configuration configmap

### PR DESCRIPTION
**Pull request by bokristoffersson**
Configuration in Curity might be done using many xml-files. By adding a configurationExtraConfigMap that support many items it is possible to configure the server using more than one file. All files specified will be mounted in etc/init folder in the container.
In the helm-values.yaml file:

configurationExtraConfigMap: extra-configmap-conf  
configurationExtraConfigMapItemNames:
        - admin-user.xml
        - authentication.xml
        - base.xml
        - environment.xml
        - facilties.xm
        - oauth.xml
        - user-managment.xml

### CODE REVIEW

**Review by delhage — APPROVED** [review]
LGTM

**Comment by anestos** [comment]
I would prefer to deprecate `configurationConfigMapItemName`. 
Until it is removed we can combine whatever is set in that old value with a new one `configurationConfigMapItemNames` (or `configurationCofigMapItemNameArray` for example). 
That way we don't introduce the `configurationExtraConfigMap` which seems a bit unnecessary imo. 

Same change would be welcomed in the `configurationSecret`

**Comment by bokristoffersson** [comment]
Thanks for the feedback. This PR is now adding support for mounting several files from both configmaps and secrets. 
`curity:
  config:
    configuration:
      - configMapRef:
          name: cfg-configmap-name
          items:
            - key: item-1
              path: item-1.xml
            - key: item-2
              path: item-2.xml
      - secretRef:
          name: cfg-secret
          items:
            - key: scrt-1
              path: scrt-1.xml
            - key: scrt-2
              path: scrt-2.xml
      - secretRef:
          name: cfg-secret-2
          items:
            - key: cfg
              path: configuration.xml
`

**Comment by bokristoffersson** [comment]
Thanks for the feedback. This PR is now adding support for mounting several files from both configmaps and secrets. 
`curity:
  config:
    configuration:
      - configMapRef:
          name: cfg-configmap-name
          items:
            - key: item-1
              path: item-1.xml
            - key: item-2
              path: item-2.xml
      - secretRef:
          name: cfg-secret
          items:
            - key: scrt-1
              path: scrt-1.xml
            - key: scrt-2
              path: scrt-2.xml
      - secretRef:
          name: cfg-secret-2
          items:
            - key: cfg
              path: configuration.xml
`

File: idsvr/templates/deployment-admin.yaml [hunk 1]
```diff
@@ -171,6 +189,28 @@ spec:
           configMap:
             name: {{ .Values.curity.config.configurationConfigMap }}
         {{- end }}
+        {{- range $conf := .Values.curity.config.configuration }}
+        {{- if $conf.configMapRef }}
+        - name: {{ $conf.configMapRef.name }}-volume
+          configMap:
+            name: {{ $conf.configMapRef.name | quote }}
```
[the remark this answers is not in the document]
Comment (bokristoffersson) [reply idsvr/templates/deployment-admin.yaml:196]: Well, I have no reason for it so I will remove it.

**Pull request merged**